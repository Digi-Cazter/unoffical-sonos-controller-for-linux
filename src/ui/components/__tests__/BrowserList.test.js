import { h } from 'preact';
import { BrowserList } from '../BrowserList';
import BrowserListItem from '../BrowserListItem';
import { deep } from 'preact-render-spy';

jest.mock('../BrowserListItem');
BrowserListItem.mockReturnValue(<p />);

describe('BrowserList', () => {
    it('renders and matches snapshot', () => {
        const props = {
            currentState: {
                items: [
                    {
                        id: 1
                    }
                ],
                title: 'My List',
                sournce: 'source'
            },
            history: []
        };

        const context = deep(<BrowserList {...props} />);
        expect(context.output()).toMatchSnapshot();

        const [firstCall] = BrowserListItem.mock.calls;

        expect(firstCall[0].model).toMatchObject({
            id: 1
        });
    });
});
