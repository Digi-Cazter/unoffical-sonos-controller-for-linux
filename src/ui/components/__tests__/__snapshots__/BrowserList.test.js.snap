// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserList renders and matches snapshot 1`] = `
<div
  id="music-sources-container"
  onScrollCapture={[Function bound _onScroll]}
>
  <h4>My List</h4>
  <ul id="browser-container">
    <div
      onScroll={[Function anonymous]}
      class="scrollcontainer"
    >
      <div style="position:relative; overflow:hidden; width:100%; min-height:100%; height:100px;">
        <div style="position:absolute; top:0; left:0; height:100%; width:100%; overflow:visible; top:0px;"></div>
      </div>
    </div>
  </ul>
</div>
`;
